smart-contracts/tests/Challenge_7.t.sol

```diff
@@ -0,0 +1,53 @@
+// SPDX-License-Identifier: Unlicense
+pragma solidity >=0.6.0 <0.9.0;
+
+import 'forge-std/Test.sol';
+
+import '@smart-contracts/Challenge_7.sol';
+
+contract Challenge7 is Test {
+  using Strings for uint256;
+  Contract c;
+  string baseURI = 'ipfs://QmeSjSinHpPnmXmspMjwiXyN6zS4E9zccariGR3jxcaWtq/';
+  function setUp() public {
+    c = new Contract('NFT Example', 'NFTEX', 100);
+    c.setBaseURI(baseURI);
+  }
+  // /// @notice Checks that the tokenURI is properly constructed
+  // /// @param _id The token id
+  function testTokenUri(uint256 _id) public {
+    assertEq(c.tokenURI(_id), string.concat(baseURI, _id.toString()));
+  }
+  /// @notice Test minting 1 from Externally Owned Account (EOA)
+  function testMintEOA() public {
+    console.log('Minting from EOA...');
+    uint256 initialSupply = c.totalSupply();
+    uint256 amount = 1;
+    /// @dev Specify the msg.sender as minter
+    vm.startPrank(msg.sender);
```

Comment:
What actually is msg.sender in this case? Isnt this by default the same as who setup the contract, and thus this is not needed? 

If you want to educate them about addresses, what would make a good test is checking the assignment of NFT ownership. So, during setup, you generate a couple of addresses E.G. Alice and Bob, and then reuse these during a test to see if it was assigned properly. And also, (OR) test with address fuzzing

Reply:
That is initially what I also assumed but this does not appear to be the case, so I had to use the prank here. Of course, this issue can indeed also be circumvented here by declaring Alice, Bob addresses / fuzzing which is a great idea. Regardless, I think the behaviour here is quite strange - let's discuss this on Wednesday together, and also the compilation issue!